backend/store/etcd/watcher.go

```diff
@@ -62,25 +71,25 @@ func (w *Watcher) Result() <-chan store.WatchEvent {
 
 func (w *Watcher) start() {
 	// Define the client options for this watcher
-	opts := []clientv3.OpOption{clientv3.WithCreatedNotify()}
+	baseOpts := []clientv3.OpOption{clientv3.WithCreatedNotify(), clientv3.WithPrevKV()}
 	if w.recursive {
-		opts = append(opts, clientv3.WithPrefix())
+		baseOpts = append(baseOpts, clientv3.WithPrefix())
+	}
+	opts := make([]clientv3.OpOption, len(baseOpts))
+	copy(opts, baseOpts)
+	if w.revision != 0 {
+		opts = append(opts, clientv3.WithRev(w.revision))
 	}
-	opts = append(opts, w.opts...)
 
 	// Create a channel to be notified if the watch channel is closed
 	watchChanStopped := make(chan struct{})
 
-	// Create a waitgroup that will be used make sure we do not write to a closed
-	// resultChan
-	var resultChanWG sync.WaitGroup
-
 	// Create a cancellable context for our watcher
 	ctx, cancel := context.WithCancel(w.ctx)
 
 	// Start the watcher
-	logger.Debugf("starting a watcher for key %s", w.key)
-	w.watch(ctx, opts, watchChanStopped, &resultChanWG)
+	w.logger.Debug("starting a watcher")
```

Comment:
Does the watcher key not provide any helpful context for the debug logs?

Reply:
Seconded, this is not really useful logger output without the watcher key.

Reply:
The key is now part of a field within the logrus entry, see [line 60](https://github.com/sensu/sensu-go/pull/3181/files#diff-2b8fe086ea46203124c5b444c0be9787R60) above, so now all log entries indicate the key without having to format it every time!